src/viam/ur/module/test.cpp

```diff
@@ -13,8 +13,57 @@ Eigen::VectorXd makeVector(std::vector<double> data) {
     return Eigen::Map<Eigen::VectorXd>(data.data(), boost::numeric_cast<Eigen::Index>(data.size()));
 }
 
+// trajectory_sample_point
 }  // namespace
 
+BOOST_AUTO_TEST_CASE(test_sampling_func) {
+    // test a random set of samples
+    {
+        std::vector<trajectory_sample_point> test_samples = {};
+        const double test_duration_sec = 5;
+        const double test_freq_hz = 5;
+        // we are not testing the behavior of the trajectory generation, so we create a simple lambda function.
+        // this allows us to get a good idea of what sampling_func is doing
+        sampling_func(test_samples, test_duration_sec, test_freq_hz, [](const double t, const double step) {
+            return trajectory_sample_point{{t, 0, 0, 0, 0, 0}, {t * step, 0, 0, 0, 0, 0}, boost::numeric_cast<float>(step)};
+        });
+        BOOST_CHECK_EQUAL(test_samples.size(), 26);
+    }
+    // check for durations smaller than the sampling frequency
+    {
+        std::vector<trajectory_sample_point> test_samples = {};
+        const double test_duration_sec = 1;
+        const double test_freq_hz = 0.5;  // 1 sample every 2 seconds
```

Comment:
not sure if this is necessarily a good combo for checking 'durations smaller than sampling frequency' because it hits the lower bounds for sampling size. might be more accurate to say this 'checks for minimum sampling bounds on valid input'.